### ISSUE TITLE
[OP] Refactor of BatchNorm Op

### ISSUE THREAD

**Issue by antinucleon** [dmlc/mxnet#1027]
- [ ] Verify ```eps``` setting, and other recent change
- [ ] Make ```var``` in same format as CuDNN
- [ ] Update pretrained model

Is there any thing I missed?
@piiswrong @winstywang @mavenlin

**Comment by piiswrong** [comment]
cudnnbn also doesn't output no_affine. Why is our version output it? Is is necessary?

**Comment by tqchen** [comment]
It is an intermediate result holder which is not a must, recomputation can always be done. If the out_no_affine is not needed, then maybe input is needed, or we always fix gamma, and add a BiasAdd Operator

**Comment by piiswrong** [comment]
I think recompute is better as it saves a lot of memory
